### PR TITLE
Enable 1-year reports

### LINKED ISSUE #2747: Giskard: Reports for One year Unavailable
## 🐛 Bug Report
When using the new Android app, it is impossible to select a time frame of one year for reports.

## Context
N/A

Build/Branch:
Phone OS: Android 8.0 (non-rooted)
Phone model: Tested and verified on Motorola g6

## Expected behavior
A user should be able to generate a report within the mobile app for one year in length; selecting the start date and end date. 

## To reproduce
-Open reports in the app
-Go back to select date one year before today's date (tested 26 July, so attempts were made to select 26 July of 2017)
-Calendar only goes back to 1 August 2017 (at the time of this writing)

## Suggestions
N/A

### PR DESCRIPTION
**Pull request by dshr**
## What's this?
A fix to a simple off-by-one bug that prevented users from generating 1 year reports.

### Relationships
Closes #2747

## Why do we want this?
Our users want this.

## How is it done?
By changing the number of months to show — we need 13 to get to the beginning of the current month last year.

### Why not another way?
Can't think of another way.

### Side effects
_Keine_

## Review hints
Open reports, try to go a whole year back, and compare against `develop`.

## :squid: Permissions
Go for it!

### CODE REVIEW

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs [hunk 1]
```diff
@@ -168,18 +168,18 @@ public sealed class TheCurrentMonthProperty : ReportsCalendarViewModelTest
         public sealed class TheCurrentPageProperty : ReportsCalendarViewModelTest
         {
             [Fact, LogIfTooSlow]
-            public void IsInitializedTo11()
+            public void IsInitializedTo12()
```
Comment (willsb) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs]: `IsInitializedToTheMonthsToShowConstantMinusOne` will make this future proof.

File: Toggl.Foundation.MvvmCross/ViewModels/ReportsCalendarViewModel.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace Toggl.Foundation.MvvmCross.ViewModels
     [Preserve(AllMembers = true)]
     public sealed class ReportsCalendarViewModel : MvxViewModel
     {
-        private const int monthsToShow = 12;
+        private const int monthsToShow = 13;
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/ReportsCalendarViewModel.cs]: Expose this as a public const. It'll be needed in the future and will make tests less magical

**Comment by dshr** [comment]
Pinging @pe0ny for some copy:

File: Toggl.Daneel/Services/DialogService.cs [hunk 3]
```diff
@@ -3,9 +3,12 @@
 using System.Reactive;
 using System.Reactive.Disposables;
 using System.Reactive.Linq;
+using Toggl.Daneel.ViewControllers;
```
Comment (willsb) [inline Toggl.Daneel/Services/DialogService.cs]: Is this even needed?

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs [hunk 4]
```diff
@@ -168,18 +168,18 @@ public sealed class TheCurrentMonthProperty : ReportsCalendarViewModelTest
         public sealed class TheCurrentPageProperty : ReportsCalendarViewModelTest
         {
             [Fact, LogIfTooSlow]
-            public void IsInitializedTo11()
+            public void IsInitializedToTheMonthsToShowConstantMinusOne()
             {
-                ViewModel.CurrentPage.Should().Be(11);
+                ViewModel.CurrentPage.Should().Be(ReportsCalendarViewModel.MonthsToShow);
```
Comment (willsb) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs]: This is not testing the correct thing

File: Toggl.Giskard/Services/DialogService.cs [hunk 5]
```diff
@@ -4,6 +4,7 @@
 using System.Reactive.Disposables;
 using System.Reactive.Linq;
 using Android.Support.V7.App;
+using Android.Widget;
```
Comment (willsb) [inline Toggl.Giskard/Services/DialogService.cs]: Is this change needed?

File: Toggl.Foundation.MvvmCross/ViewModels/Reports/ReportsViewModel.cs [hunk 6]
```diff
@@ -168,8 +168,8 @@ public bool IsCurrentWeek
             this.dialogService = dialogService;
             this.intentDonationService = intentDonationService;
 
-            calendarViewModel = new ReportsCalendarViewModel(timeService, dataSource, intentDonationService);
-
+            calendarViewModel = new ReportsCalendarViewModel(timeService, dialogService, dataSource, intentDonationService);
+
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/Reports/ReportsViewModel.cs]: Whitespace

File: Toggl.Foundation.MvvmCross/ViewModels/ReportsCalendarViewModel.cs [hunk 7]
```diff
@@ -43,6 +45,7 @@ public sealed class ReportsCalendarViewModel : MvxViewModel
         private bool isInitialized;
         private CalendarMonth initialMonth;
         private CompositeDisposable disposableBag;
+
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/ReportsCalendarViewModel.cs]: Why the space?
